packages/instrumenter/test/unit/mutant-placers/expression-mutant-placer.spec.ts

```diff
@@ -75,10 +75,11 @@ describe(expressionMutantPlacer.name, () => {
         ['foo?.bar()', (p) => p.isOptionalMemberExpression() && types.isIdentifier(p.node.property, { name: 'bar' })],
         ['foo.bar?.baz', (p) => p.isMemberExpression() && types.isIdentifier(p.node.property, { name: 'bar' })],
         ['foo?.bar.baz', (p) => p.isOptionalMemberExpression() && types.isIdentifier(p.node.property, { name: 'bar' })],
+        ['foo?.bar!.baz', (p) => p.isTSNonNullExpression()],
       ];
       falsePointers.forEach(([js, query]) => {
         it(`should not allow placing in \`bar\` of \`${js}\``, () => {
-          const path = findNodePath(parseJS(js), query);
+          const path = findNodePath(parseTS(js), query);
           expect(expressionMutantPlacer.canPlace(path)).false;
```

Comment:
Nice addition! Could we also add a bon-null assertion to the integration test and update the snapshot?

Reply:
Did that ✅